examples/transaction-import/pti_tools.py

```diff
@@ -0,0 +1,106 @@
+def sign_payload(client_id, key, payload, compact=True):
+    public_key = jwk.JWK()
+    public_key.import_key(**json_decode(key.export_public()))
+    jwstoken = jws.JWS(payload)
+    jwstoken.add_signature(key, None,
+                            json_encode({"alg": "RS512",
+                                         "cid": client_id,
+                                         "kid": public_key.thumbprint()}), None)
+    signed_payload = jwstoken.serialize(compact)
+    return signed_payload
+
+def get_signature(client_id, private_key, url, method, data=None, content_type="application/json", date=get_http_gmt()):
+    method = "GET"
+    if data:
+        method = "POST"
+
+    log.debug(f'Data: {data}')
+
+    payload = method + "\n"
+    if method == "POST":
+        payload = payload + get_content_sha256(data.encode('UTF-8')) + "\n"
+        payload = payload + 'content-type:' + content_type + "\n"
+    else:
+        payload + "\n\n"
+
+    payload = payload + 'date:' + date + '\n'
+    payload = payload + f'x-pti-client-id:{client_id}' + '\n'
+    payload = payload + urllib.parse.urlparse(url).path
+    signature = sign_payload(client_id, private_key, payload.encode('UTF-8'))
+
+    log.debug(f'Payload: {payload}')
+    log.debug(f'Signature (JWS): {signature}')
+
+    return signature
+
+def make_signed_request(client_id, request_id, key, url, method, data=None, content_type="application/json"):
+    date = get_http_gmt()
+    signature = get_signature(client_id, key, url, method, data, content_type, date)
+    if method == "POST":
+        resp = requests.post(url, data=data.encode('UTF-8'), headers={"Content-Type": content_type, "Date": date, "x-pti-signature": signature, "x-pti-client-id": client_id, "x-pti-request-id": request_id })
+    else:
+        resp = requests.get(url, headers={"Date": date, "x-pti-signature": signature,  "x-pti-client-id": client_id, "x-pti-request-id": request_id})
+
+    return resp
+
+def decrypt_verify_and_get_payload(private_encryption_key, public_signature_key, content):
+      # decrypt
+      jwetoken = jwe.JWE()
+      jwetoken.deserialize(content, key=private_encryption_key)
+      log.debug(f'Payload correctly decrypted using our private key')
+      if json.loads(jwetoken.objects['protected'])['alg'] != 'RSA-OAEP-256' or json.loads(jwetoken.objects['protected'])['enc'] != 'A256CBC-HS512':
+        raise Exception('Unsupported encryption') # we have to check those, otherwise other encryptions can be used as an attack vector
+
+      log.debug(f'JWE payload: {jwetoken.payload}')
+      # verify
+      jwstoken = jws.JWS()
+      jwstoken.deserialize(jwetoken.payload)
+
+      if json.loads(jwstoken.objects['protected'])['alg'] != 'RS512':
+        raise Exception('Unsupported signature algorithm') # we have to check, otherwise other algorithms can be used as an attack vector
+
+      jwstoken.verify(public_signature_key) # if there's no exception, the signature is valid
+      return jwstoken.payload
```

Comment:
I did not do that here, but we should have a common file for python stuff and not copy paste this one around